src/node_http2.cc

```diff
@@ -1558,18 +1558,10 @@ void Http2Session::SendPendingData() {
 
   chunks_sent_since_last_write_++;
 
-  // DoTryWrite may modify both the buffer list start itself and the
-  // base pointers/length of the individual buffers.
-  uv_buf_t* writebufs = *bufs;
-  if (stream_->DoTryWrite(&writebufs, &count) != 0 || count == 0) {
-    // All writes finished synchronously, nothing more to do here.
-    ClearOutgoing(0);
-    return;
-  }
-
-  WriteWrap* req = AllocateSend();
-  if (stream_->DoWrite(req, writebufs, count, nullptr) != 0) {
-    req->Dispose();
+  StreamWriteResult res =
+      static_cast<StreamBase*>(stream_)->Write(*bufs, count);
```

Comment:
StreamResource is-a StreamBase so Write() is available, isn't it?  The static_cast looks superfluous to me.

Reply:
> StreamResource is-a StreamBase

Unfortunately, it’s the reverse situation. Maybe `StreamBase` is not the *best* name for this… :/

Anyway, I’ve added a `underlying_stream()` helper to `Http2Session` and `TLSWrap` that makes the cast transparent, so that shouldn’t be an issue for niow.